### PR TITLE
PropertyInfo Fixes (Reduce Cognitive load, use FullNamespace for core functions...etc)

### PR DESCRIPTION
**Pull request by NikoGrano**
| Q             | A
| ------------- | ---
| Branch?       | 4.2
| Bug fix?      | no
| New feature?  | no 
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -
| Previous PR | https://github.com/symfony/symfony/pull/30542

### CODE REVIEW

**Review by dunglas — APPROVED** [review]

**Comment by dunglas** [comment]
Why not, but it will create more merge conflicts when merging 3.4 (because 3.4 still supports PHP 5).

**Comment by Niko9911** [comment]
Is this mandatory to be merged into 3.4? As for example `??` IS NOT supported in php 5.x?

**Comment by dunglas** [comment]
@Niko9911 it cannot be merged in 3.4. The problem is:

* we merge bug fixes in 3.4 (because it's a LTS version)
* we regularly merge 3.4 in newer branches such as  4.2

With this can of patches, the number of merge conflicts when merging bug fixes (that must be written in PHP 5) in 4.2+ will increase.

TLDR: we'll be able to use PHP7 only syntax only when we'll stop supporting Symfony 3 LTS.

**Comment by fabpot** [comment]
Thanks for proposing. I'm going to close as we don't merge such PRs about cosmetic changes. If we want to do such changes (like `list()` to `[]`), they must be part of PHP-CS-Fixer so that we can automate them completely on the whole code base.

**Comment by Niko9911** [comment]
Code-style or not, that code doesn't not honor strict checking, best practices. This component needs anyways rewrite as it's not working as it should.

**Comment by dunglas** [comment]
> This component needs anyways rewrite as it's not working as it should.

What do you mind?

**Comment by Niko9911** [comment]
#30562
